### ISSUE TITLE
factorial(x::Vector{Int64}) StackOverflow

### ISSUE THREAD

**Issue by anandijain** [JuliaMath/SpecialFunctions.jl#304]
```julia
julia> SpecialFunctions.factorial([3])
ERROR: StackOverflowError:
Stacktrace:
 [1] factorial(x::Vector{Int64})
   @ SpecialFunctions ~/.julia/packages/SpecialFunctions/LC8dm/src/gamma.jl:887
```

I know this is improper usage, but SO is unexpected.

**Comment by ararslan** [comment]
Yeah, it looks like https://github.com/JuliaMath/SpecialFunctions.jl/blob/master/src/gamma.jl#L887-L891 is incorrect and simply enters infinite recursion, since we're doing `import Base: factorial` then defining that to call `Base.factorial`, and the function has no type restriction on the argument.

**Comment by devmotion** [comment]
This is fixed in SpecialFunctions >= 2.0.0 which removed the `Base.factorial` overloads:
```julia
julia> SpecialFunctions.factorial([3])
ERROR: MethodError: no method matching factorial(::Vector{Int64})
Closest candidates are:
  factorial(::Union{Int64, UInt64}) at ~/.julia/juliaup/julia-1.7.3+0~x64/share/julia/base/combinatorics.jl:27
  factorial(::Union{Int16, Int32, Int8, UInt16, UInt32, UInt8}) at ~/.julia/juliaup/julia-1.7.3+0~x64/share/julia/base/combinatorics.jl:33
  factorial(::BigInt) at ~/.julia/juliaup/julia-1.7.3+0~x64/share/julia/base/gmp.jl:645
  ...
Stacktrace:
 [1] top-level scope
   @ REPL[3]:1
```
